modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/JsonProcessor.java

```diff
@@ -117,17 +140,23 @@ public JsonProcessor create(Map<String, Processor.Factory> registry, String proc
             String field = ConfigurationUtils.readStringProperty(TYPE, processorTag, config, "field");
             String targetField = ConfigurationUtils.readOptionalStringProperty(TYPE, processorTag, config, "target_field");
             boolean addToRoot = ConfigurationUtils.readBooleanProperty(TYPE, processorTag, config, "add_to_root", false);
+            boolean addToRootRecursiveMerge = ConfigurationUtils.readBooleanProperty(TYPE, processorTag, config,
+                "add_to_root_recursive_merge", false);
 
             if (addToRoot && targetField != null) {
                 throw newConfigurationException(TYPE, processorTag, "target_field",
                     "Cannot set a target field while also setting `add_to_root` to true");
             }
+            if (addToRoot == false && addToRootRecursiveMerge) {
+                throw newConfigurationException(TYPE, processorTag, "add_to_root_recursive_merge",
```

Comment:
maybe also add a unit test to see that this fails?

Reply:
done